src/doom/p_enemy.c

```diff
@@ -235,34 +235,26 @@ boolean P_CheckMissileRange (mobj_t* actor)
 
     dist >>= FRACBITS;
 
-    if (actor->type == MT_VILE)
-    {
-	if (dist > 14*64)	
-	    return false;	// too far away
-    }
-	
-
-    if (actor->type == MT_UNDEAD)
-    {
-	if (dist < 196)	
-	    return false;	// close for fist attack
-	dist >>= 1;
-    }
+    // generalization of the Arch Vile's different attack range
+    if (actor->info->maxattackrange > 0 && dist > actor->info->maxattackrange)
```

Comment:
Please preserve the original code as much as possible, i.e same structure and indentation.

Reply:
Sorry, I was following the same coding conventions as the code around it.
Do you mean that I should keep the statement surrounded in braces even though it's only 1 line?

Reply:
Yes, please try to keep the code as original as possible - even if this means being inconcistent if the original code is already inconsistent.

Reply:
Previous code wasn't inconsistent (it was doing 2 things, so it needed the braces).
But sure, I'll keep the braces.